lib/std/os/linux/syscalls.zig

```diff
@@ -0,0 +1,3487 @@
+// This file is automatically generated.
+// See tools/generate_linux_syscalls.zig for more info.
+pub const SYS = switch (@import("builtin").cpu.arch) {
```

Comment:
If this were a `fn` which returned the enum type instead:
- zls would autocomplete it for the current architecture
- applications could use the syscall numbers for a different architecture (dont know the usecase)

For example:
```zig
pub fn NewSYS(comptime arch: @TypeOf(@import("builtin").cpu.arch)) type {
   return switch (arch) {
      // rest of code
   };
}

pub const SYS = NewSYS(@import("builtin").cpu.arch);
```

Reply:
My first versions had separate `enum`s for each arch, and a global `SYS` that switched for the correct enum. This is similar to how it is currently with `linux.arch_bits`. The only use of different variables would be checking the syscall numbers in a seccomp program.

When you say:
 > zls would autocomplete it for the current architecture

is this something that can be fixed in zls?